docs/001_develop/03_client-capabilities/023_custom-components/03_state-management.mdx

```diff
@@ -183,102 +230,74 @@ export class MyButton extends GenesisElement {
   clickCount = 0; // Internal property, not linked to an attribute
 
   handleClick() {
-    this.clickCount++; // Update internal state
+    this.clickCount += 1; // Update internal state
     console.log(`Button clicked ${this.clickCount} times`);
   }
 }
 ```
 
-- They can be declared using @attr which provides reflection between the JavaScript property and the HTML attribute. Changes to the attribute update the property, and vice versa.
+- Properties can be declared using `@attr` which provides reflection between the JavaScript property and the HTML attribute. Changes to the attribute will update the property, and vice versa.
 
 ```typescript
 export class MyButton extends GenesisElement {
   @attr label = "Click Me"; // Declared as a linked property
 }
 ```
 
-- The _:_ symbol is used for property binding, allowing you to bind values directly to DOM properties instead of HTML attributes. This is useful when you want to bind complex data types or frequently changing properties that shouldn’t be reflected as attributes in the DOM.
+- The **_:_** symbol is used for property binding, allowing you to bind values directly to DOM properties instead of HTML attributes. This is useful when you want to bind complex data types or frequently changing properties that shouldn’t be reflected as attributes in the DOM.
 
 See `clickCount`
 
 ```typescript
-
 @customElement({
-  name: 'my-button',
-  template: html`
-    <button> :clickCount="${x => x.clickCount}"></button>
-`;,
+  name: "my-button",
+  template: html<MyButton>`<button
+    :clickCount="${(x) => x.clickCount}" // :clickCount property binding dynamically updates the clickCount property of the <button> element.
+    @click="${(x) => x.handleClick()}"
+  >
+    Clicked ${(x) => x.clickCount} times
+  </button>`,
   styles,
 })
-
 export class MyButton extends GenesisElement {
-    @attr label = "Click Me"; // Linked attribute for label
-    clickCount = 0; // Internal property, not linked to an attribute
+  clickCount = 0; // Internal property, not reflected as an attribute
 
-    handleClick() {
-        this.clickCount++; // Update internal state
-        console.log(`Button clicked ${this.clickCount} times`);
-    }
+  handleClick() {
+    this.clickCount += 1; // Update internal property
+    console.log(`Button clicked ${this.clickCount} times`);
+  }
 }
-
 ```
```

Comment:
Maybe add a setence after this saying here clickCount is just a number and could be an attribute, but binding to a property is useful if you need to set complex data types such as arrays and objects